chisel/src/session_source.rs

```diff
@@ -100,6 +100,22 @@ impl SessionSourceConfig {
 
         match solc_req {
             SolcReq::Version(version) => {
+                // We now need to verify if the solc version provided is supported by the evm
+                // version set. If not, we need to install the latest version of
+                // solc that is supported by the evm version.
+                // 1. Do we need solc 0.8.18 or higher?
+                let needs_post_merge_solc = self.foundry_config.evm_version >= EvmVersion::Paris;
+                // 2. Check if the version provided is less than 0.8.18 and override it,
+                // or leave it as-is if we don't need a post merge solc version or the version we
+                // have is good enough.
+                if let v = needs_solc_18 && version < Version::new(0, 8, 18) {
+                    // If we do need a new version, install 0.8.19 (Paris)
+                    // NOTE: This needs to be bumped to 0.8.20 once we upgrade all tooling to Shanghai.
+                   Version::new(0, 8, 19)
```

Comment:
Could also probably also add a shell warning here saying that the provide solc version has been overriden

Reply:
yeh, user input should take precedence, would even prefer an error if conflicting user input is provided.

Reply:
^ agree

Reply:
aight, changed this to instead bail if the version provided is not recent enough: https://github.com/foundry-rs/foundry/pull/4949/commits/0eda884a042befca1aa5fd484ddf36cc2cce2d4f